core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/RedHat/setup.yml

```diff
@@ -23,13 +23,15 @@
     - not custom_repository_url
     - inventory_hostname in target_repository_hostnames
 
-- name: Include Helm binary installation 
-  include_role:
-    name: helm
-    tasks_from: install-bin
-
-- name: Include helm repository creation
-  include_tasks: "create-helm-repo.yml"
+- name: Helm binary installation and Helm repository creation
+  block:
+    - name: Include Helm binary installation 
+      include_role:
+        name: helm
+        tasks_from: install-bin
+    - name: Include Helm repository creation
+      include_tasks: "create-helm-repo.yml"
+  when: inventory_hostname in groups['kubernetes_master'][0]
```

Comment:
Same as for Debian.
Also this code is not specific to either Debian or Redhat or in fact any distro since helm binary is statically linked. I'd suggest putting that code in a single place outside of Redhat or Debian specific code.